lib/msal-node/test/client/ManagedIdentityApplication.spec.ts

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ManagedIdentityApplication } from "../../src/client/ManagedIdentityApplication";
+import { ManagedIdentityConfiguration } from "../../src/config/Configuration";
+import { ManagedIdentityRequestParams } from "../../src/request/ManagedIdentityRequestParams";
+import {
+    DEFAULT_MANAGED_IDENTITY_AUTHENTICATION_RESULT,
+    MANAGED_IDENTITY_CLIENT_ID,
+    MANAGED_IDENTITY_RESOURCE,
+} from "../test_kit/StringConstants";
+
+import { ManagedIdentityTestUtils } from "../test_kit/ManagedIdentityTestUtils";
+import {
+    DEFAULT_MANAGED_IDENTITY_ID,
+    ManagedIdentityIdType,
+} from "../../src/utils/Constants";
+
+describe("ManagedIdentityApplication unit tests", () => {
+    let OLD_ENVS: NodeJS.ProcessEnv;
+    beforeAll(() => {
+        // make a copy of old environment
+        OLD_ENVS = process.env; // Restore old environment
+    });
+    afterAll(() => {
+        // restore old environment
+        process.env = OLD_ENVS;
+    });
+
+    const managedIdentityRequestParams: ManagedIdentityRequestParams = {
+        resource: MANAGED_IDENTITY_RESOURCE,
+    };
+
+    const nonSystemAssignedConfig: ManagedIdentityConfiguration = {
+        system: {
+            networkClient:
+                ManagedIdentityTestUtils.getManagedIdentityNetworkClient(
+                    MANAGED_IDENTITY_CLIENT_ID
+                ),
+        },
+        managedIdentityIdParams: {
```

Comment:
In hindsight, I think this would be better named as "managedIdentityConfig"

Reply:
I don't think this makes sense because the type is already ManagedIdentityConfiguration

Reply:
Let's review the public API in a meeting.